doc/contribute/design/index.rst

```diff
@@ -56,14 +58,181 @@ Encryption keys are automatically refreshed periodically (in line with https://d
+        s_waiting(state:=WAITING_FOR_FINAL)
+        s_established(state:=ESTABLISHED)
+
+        s_generate[Generate key]
+        s_queue_new(outgoing:=M)
+        s_emit_kei[Emit key_exchange_init]
+        s_queue_new_init(Outgoing:=M')
+        s_emit_ker[Emit key_exchange_response]
+        s_use_key[Use their key]
+        s_queue_new_waiting(Outgoing:=M')
+        s_emit_kef[Emit key_exchange_final]
+        s_dual_kei{Which has priority?}
+
+        s_unknown =="receive|send"==> s_inactive
+        s_inactive =="send(M)"==> s_queue_new
+        s_queue_new ==> s_generate
+
+        %% They're trying to talk to us, but we're not in the right state! Help
+        %% them start over
+        s_inactive --"receive(response)|receive(final)"--> s_generate
+
+        s_generate ==> s_emit_kei
+
+        s_emit_kei ==> s_initiated
+
+        s_initiated --"send(M')"--> s_queue_new_init
+        s_queue_new_init --> s_emit_kei
+        s_initiated --> s_emit_kei
+
+        s_initiated =="receive(response)"==> s_emit_kef
+        s_established --"receive(response)"--> s_emit_kef
+        s_emit_kef ==> s_established
+        
+        s_initiated --"receive(init)"--> s_dual_kei
+        s_dual_kei --I win--> s_emit_kei
+        s_dual_kei --They win--> s_use_key
+        s_use_key ==> s_emit_ker
+
+        s_inactive =="receive(init)"==> s_use_key
+        s_emit_ker ==> s_waiting
+        
+        s_waiting --"send(M')"--> s_queue_new_waiting
+        s_queue_new_waiting --> s_emit_ker
+
+        s_waiting --"receive(init)"--> s_emit_ker
+        s_waiting --"receive(response)"--> s_waiting
+        s_waiting =="receive(final)"====> s_established
+
+        s_initiated-."close()"....-> s_inactive
+        s_waiting-."close()"....-> s_inactive
+        s_established-."close()"....-> s_inactive
+
+This doesn't yet deal with key rotation, and I believe we need an establishment-attempt nonce to prevent replay attacks that could re-establish an old, overused key. While the summary above says we start a fresh key-exchange protocol, the actual implementation attempts to do this in parallel with encrypted messages over an existing channel. If we build a protocol which can reliably deal with legitimate reconnection attempts, it would be preferable to re-use that for key rotation - deliberately close an existing connection and start fresh with a new channel - rather than building an additional rotation protocol.
+
+An open question here is whether we need to handle channel closures and re-opening. If we don't, then we can consider all dropped messages here as malicious DoS, but perhaps simplify the protections. One option is to avoid ever closing a channel - if a node has opened a channel to another node, it remains communicating with them forever. Another is to determine a point after which a channel can be safely closed - if the channel is used only for consensus and not forwarding, it may be safe to close after a node's retirement.
```

Comment:
So I'm not 100% on this, but I see 2 reasons why making these channels consensus-only doesn't give us a win:

- The point where we can close a channel (we're done talking to them, and they should never try to talk to us), is later than anything that we currently track. It is when they have ACK'd global commit of their own retirement. At any time before they know their retirement _is globally committed_, they may become candidate and still ask us for votes + require us (and other nodes) to help them rediscover the global commit index.
- I think we should implement key rotation as fresh key exchange, which means we need to handle multiple key exchanges  from the same node, which means we have at least once case of legitimate channel closure and re-opening ("they want to use a new key"). It's possible we can somehow design key rotation as a separate protocol over an existing channel, but I don't see an easy way to do so and it would mean designing+verifying+maintaining 2 protocols.